prov/gni/src/gnix_vc.c

```diff
@@ -1220,37 +1152,96 @@ int _gnix_vc_disconnect(struct gnix_vc *vc)
 	return FI_SUCCESS;
 }
 
-int _gnix_vc_add_to_wq(struct gnix_vc *vc)
+int _gnix_vc_schedule(struct gnix_vc *vc)
 {
 	struct gnix_nic *nic = vc->ep->nic;
-	struct gnix_work_req *work_req;
 
-	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
+	if (!_gnix_test_and_set_bit(&vc->flags, GNIX_VC_FLAG_SCHEDULED)) {
+		GNIX_INFO(FI_LOG_EP_CTRL, "Scheduled VC (%p)\n", vc);
```

Comment:
This seems like it could be confusing when debugging, since it's not *scheduled* yet.  I might change it to *scheduling* or move it to after the `fastlock_release`.